hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/BlocklistAccountCreator.java

```diff
@@ -0,0 +1,169 @@
+
+import com.google.protobuf.ByteString;
+import com.hedera.node.app.service.evm.utils.EthSigsUtils;
+import com.hedera.node.app.service.mono.context.annotations.CompositeProps;
+import com.hedera.node.app.service.mono.context.properties.PropertySource;
+import com.hedera.node.app.service.mono.exceptions.NegativeAccountBalanceException;
+import com.hedera.node.app.service.mono.ledger.accounts.AliasManager;
+import com.hedera.node.app.service.mono.ledger.accounts.HederaAccountCustomizer;
+import com.hedera.node.app.service.mono.ledger.backing.BackingStore;
+import com.hedera.node.app.service.mono.ledger.ids.EntityIdSource;
+import com.hedera.node.app.service.mono.legacy.core.jproto.JEd25519Key;
+import com.hedera.node.app.service.mono.legacy.core.jproto.JKey;
+import com.hedera.node.app.service.mono.state.migration.HederaAccount;
+import com.hedera.node.app.service.mono.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.Key;
+import com.hederahashgraph.api.proto.java.KeyList;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.bouncycastle.asn1.sec.SECNamedCurves;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.crypto.params.ECPublicKeyParameters;
+import org.bouncycastle.util.encoders.Hex;
+
+@Singleton
+public class BlocklistAccountCreator {
+    public static final String BLOCKLIST_ACCOUNT_MEMO = "Account is blocked";
+    private static final int ZERO_BALANCE = 0;
+    private final Supplier<HederaAccount> accountSupplier;
+    private final EntityIdSource ids;
+    private final BackingStore<AccountID, HederaAccount> accounts;
+    private final Supplier<JEd25519Key> genesisKeySource;
+    private final PropertySource properties;
+    private final AliasManager aliasManager;
+    private JKey genesisKey;
+    private final List<HederaAccount> blockedAccountsCreated = new ArrayList<>();
+
+    @Inject
+    public BlocklistAccountCreator(
+            final Supplier<HederaAccount> accountSupplier,
+            final EntityIdSource ids,
+            final BackingStore<AccountID, HederaAccount> accounts,
+            final Supplier<JEd25519Key> genesisKeySource,
+            final @CompositeProps PropertySource properties,
+            final AliasManager aliasManager) {
+        this.accountSupplier = accountSupplier;
+        this.ids = ids;
+        this.accounts = accounts;
+        this.genesisKeySource = genesisKeySource;
+        this.properties = properties;
+        this.aliasManager = aliasManager;
+    }
+
+    public void ensureBlockedAccounts() {
+        final var blockedEVMAddresses = getPrivateKeyBlocklist().stream()
+                .map(Hex::decode)
```

Comment:
This can throw if the address is malformed.  One typo in the resources file and ...